lighthouse-core/audits/byte-efficiency/unused-css-rules.js

```diff
@@ -40,156 +38,32 @@ class UnusedCSSRules extends ByteEfficiencyAudit {
-   * Trims stylesheet content down to the first rule-set definition.
-   * @param {string=} content
-   * @return {string}
-   */
-  static determineContentPreview(content) {
-    let preview = (content || '')
-        .slice(0, PREVIEW_LENGTH * 5)
-        .replace(/( {2,}|\t)+/g, '  ') // remove leading indentation if present
-        .replace(/\n\s+}/g, '\n}') // completely remove indentation of closing braces
-        .trim(); // trim the leading whitespace
-
-    if (preview.length > PREVIEW_LENGTH) {
-      const firstRuleStart = preview.indexOf('{');
-      const firstRuleEnd = preview.indexOf('}');
-
-      if (firstRuleStart === -1 || firstRuleEnd === -1
-          || firstRuleStart > firstRuleEnd
-          || firstRuleStart > PREVIEW_LENGTH) {
-        // We couldn't determine the first rule-set or it's not within the preview
-        preview = preview.slice(0, PREVIEW_LENGTH) + '...';
-      } else if (firstRuleEnd < PREVIEW_LENGTH) {
-        // The entire first rule-set fits within the preview
-        preview = preview.slice(0, firstRuleEnd + 1) + ' ...';
-      } else {
-        // The first rule-set doesn't fit within the preview, just show as many as we can
-        const lastSemicolonIndex = preview.slice(0, PREVIEW_LENGTH).lastIndexOf(';');
-        preview = lastSemicolonIndex < firstRuleStart ?
-            preview.slice(0, PREVIEW_LENGTH) + '... } ...' :
-            preview.slice(0, lastSemicolonIndex + 1) + ' ... } ...';
-      }
-    }
-
-    return preview;
-  }
-
-  /**
-   * @param {StyleSheetInfo} stylesheetInfo The stylesheetInfo object.
-   * @param {string} pageUrl The URL of the page, used to identify inline styles.
-   * @return {LH.Audit.ByteEfficiencyItem}
-   */
-  static mapSheetToResult(stylesheetInfo, pageUrl) {
-    let url = stylesheetInfo.header.sourceURL;
-    if (!url || url === pageUrl) {
-      const contentPreview = UnusedCSSRules.determineContentPreview(stylesheetInfo.content);
-      url = contentPreview;
-    }
-
-    const usage = UnusedCSSRules.computeUsage(stylesheetInfo);
-    return {url, ...usage};
-  }
-
   /**
    * @param {LH.Artifacts} artifacts
-   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {LH.Artifacts.NetworkRequest[]} networkRecords
+   * @param {LH.Audit.Context} context
    * @return {Promise<ByteEfficiencyAudit.ByteEfficiencyProduct>}
    */
-  static audit_(artifacts, networkRecords) {
-    const styles = artifacts.CSSUsage.stylesheets;
-    const usage = artifacts.CSSUsage.rules;
-    const pageUrl = artifacts.URL.finalUrl;
-
-    return Promise.resolve(networkRecords).then(networkRecords => {
-      const indexedSheets = UnusedCSSRules.indexStylesheetsById(styles, networkRecords);
-      UnusedCSSRules.indexUsedRules(usage, indexedSheets);
+  static async audit_(artifacts, networkRecords, context) {
+    const unusedCssItems = await UnusedCSS.request({
+      CSSUsage: artifacts.CSSUsage,
+      URL: artifacts.URL,
+      networkRecords,
```

Comment:
this feels wrong/goes against the pattern for computed artifacts we have that we're passing in the network records directly 

can we base this off the devtoolsLog and have it request the network records instead?